sympy/tensor/indexed.py

```diff
@@ -581,7 +579,7 @@ def __new__(cls, label, range=None, **kw_args):
                 raise ValueError(filldedent("""
                     Idx range tuple must have length 2, but got %s""" % len(range)))
             for bound in range:
-                if not (bound.is_integer or abs(bound) is S.Infinity):
+                if not (bound.is_integer != False or abs(bound) is S.Infinity):
```

Comment:
Since `oo.is_integer == None` it's enough to write: `if bound.is_integer is False:`